hicexplorer/hicCompareMatrices.py

```diff
@@ -33,161 +33,54 @@ def parse_arguments(args=None):
-                        if dataFrameBins is None:
-                            dataFrameBins = dataFrameBins_
-                        else:
-                            dataFrameBins = pd.concat([dataFrameBins, dataFrameBins_], ignore_index=True)  # .append(dataFrameBins_, ignore_index=True)
-                    if dataFrameMatrix_ is not None:
-                        if dataFrameMatrix is None:
-                            dataFrameMatrix = dataFrameMatrix_
-                        else:
-                            dataFrameMatrix = pd.concat([dataFrameMatrix, dataFrameMatrix_], ignore_index=True)  # .append(dataFrameMatrix_, ignore_index=True)
-                    dataFrameBins_ = None
-                    dataFrameMatrix_ = None
-                    queue[i] = None
-                    process[i].join()
-                    process[i].terminate()
-
-                    process[i] = None
-                    thread_done[i] = True
-                elif chr_element >= len(chromosome_list) and queue[i] is None:
-                    thread_done[i] = True
-                else:
-                    time.sleep(0.1)
-                if chr_element >= len(chromosome_list):
-                    all_threads_done = True
-                    for thread in thread_done:
-                        if not thread:
-                            all_threads_done = False
-
-        hic.save_cool_pandas(args.outFileName, dataFrameBins, dataFrameMatrix)
-
-    else:
-        if args.threads:
-            print("Multiple threads are only used if the matrices are in 'cool'-format.")
-        hic1 = hm.hiCMatrix(args.matrices[0])
-        hic2 = hm.hiCMatrix(args.matrices[1])
-
-        if hic1.matrix.shape != hic2.matrix.shape:
-            exit("The two matrices have different size. Use matrices having the same resolution and created using"
-                 "the same parameters. Check the matrix values using the tool `hicInfo`.")
-
-        if hic1.chrBinBoundaries != hic2.chrBinBoundaries:
-            exit("The two matrices have different chromosome order. Use the tool `hicExport` to change the order.\n"
-                 "{}: {}\n"
-                 "{}: {}".format(args.matrices[0], hic1.chrBinBoundaries.keys(),
-                                 args.matrices[1], hic2.chrBinBoundaries.keys()))
-
-        # normalize by total matrix sum
-        hic1.matrix.data = hic1.matrix.data.astype(float) / hic1.matrix.data.sum()
-        hic2.matrix.data = hic2.matrix.data.astype(float) / hic2.matrix.data.sum()
-
-        nan_bins = set(hic1.nan_bins)
-        nan_bins = nan_bins.union(hic2.nan_bins)
-
-        if args.operation == 'diff':
-            new_matrix = hic1.matrix - hic2.matrix
-        elif args.operation == 'ratio' or args.operation == 'log2ratio':
-            hic2.matrix.data = float(1) / hic2.matrix.data
-            new_matrix = hic1.matrix.multiply(hic2.matrix)
-            # just in case
+    
+    if args.threads:
```

Comment:
Since you removed the threads option from argparse, will this still work?

Reply:
I missed to remove this from the source, thanks for the hint.